src/pubsub.c

```diff
@@ -244,20 +264,32 @@ int clientTotalPubSubSubscriptionCount(client *c) {
 /* Subscribe a client to a channel. Returns 1 if the operation succeeded, or
  * 0 if the client was already subscribed to that channel. */
 int pubsubSubscribeChannel(client *c, robj *channel, pubsubtype type) {
+    dict **pd;
     dictEntry *de;
     list *clients = NULL;
     int retval = 0;
+    unsigned int slot = 0;
 
     /* Add the channel to the client -> channels hash table */
     if (dictAdd(type.clientPubSubChannels(c),channel,NULL) == DICT_OK) {
         retval = 1;
         incrRefCount(channel);
         /* Add the client to the channel -> list of clients hash table */
-        de = dictFind(*type.serverPubSubChannels, channel);
+        if (server.cluster_enabled && type.shard) {
+            slot = keyHashSlot(channel->ptr, sdslen(channel->ptr));
+        }
+        pd = type.serverPubSubChannels(slot);
```

Comment:
Shall we use a more readable name, maybe `pubsub_dict` ?

Reply:
It is actually "pointer to dict", as we usually name `dict *` as `d`. I think `pd` is OK.

Reply:
`dict **d_ptr` is another possibility then. I agree the name `pd` is not very clear.